### PR TITLE
[Synthetics] add base-fields

### PR DESCRIPTION
**Pull request by dominiqueclarke**
Related to elastic/kibana#91062

Type of change
- Bug

## What does this PR do?

Add base fields for data_stream and dataset field mappings. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs (synthetics).
- [x] I have added an entry to my package's `changelog.yml` file.

## Author's Checklist

- [ ] Review that base fields are added correctly for each data stream

## Related issues

-

### CODE REVIEW

File: packages/synthetics/data_stream/http/fields/base-fields.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- name: dataset.type
+  type: constant_keyword
+  description: Dataset type.
+- name: dataset.name
+  type: constant_keyword
+  description: Dataset name.
+- name: dataset.namespace
+  type: constant_keyword
+  description: Dataset namespace.
+- name: '@timestamp'
```
Comment (dominiqueclarke) [inline packages/synthetics/data_stream/http/fields/base-fields.yml:19]: We defined timestamp on our `ecs.yaml` field mappings. Should I remove it from there or are both necessary?

Comment (mtojek) [reply packages/synthetics/data_stream/http/fields/base-fields.yml:19]: Good catch! You can keep it in a one place, don't have to be duplicated :)

**Review by mtojek — COMMENTED** [review]
I suppose you may need to run `elastic-package build` as well (to reflect changes in READMEs).

**Review by shahzad31 — APPROVED** [review]
LGTM !!

File: packages/synthetics/_dev/build/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Synthetics integration
+
+This integration creates and manages configuration for [Heartbeat monitors](https://www.elastic.co/guide/en/beats/heartbeat/current/configuration-heartbeat-options.html).
```
Comment (mtojek) [inline packages/synthetics/_dev/build/docs/README.md:3]: I see that the README template and generated README differ from each other (see this paragraph). Is it intentional? If not, please update this file: `packages/synthetics/_dev/build/docs/README.md` (template).

Comment (dominiqueclarke) [reply packages/synthetics/_dev/build/docs/README.md:3]: Actually, now that I'm thinking through this, we really want our README to be product driven. Our PM will decide what is on this page, and we don't believe it's useful to display all the field mappings for synthetics. for this reason, I'm going to go ahead and remove this template and revert back to the PM approved placeholder README we have now.

**Review by mtojek — APPROVED** [review]

**Pull request merged**